File: Application/src/main/java/com/shareyourproxy/api/service/MessageService.java
package com.shareyourproxy.api.service;


import com.shareyourproxy.api.domain.model.Message;

import java.util.HashMap;

import retrofit.http.Body;
import retrofit.http.DELETE;
import retrofit.http.GET;
import retrofit.http.PUT;
import retrofit.http.Path;
import rx.Observable;

/**
 * Service to get and update messages.
 */
public interface MessageService {
    @GET("/messages/{userId}.json")
    Observable<HashMap<String, Message>> getUserMessages(@Path("userId") String userId);

    @PUT("/messages/{userId}.json")
    Observable<HashMap<String, Message>> addUserMessage(
        @Path("userId") String userId, @Body HashMap<String, Message> message);

    @DELETE("/messages/{userId}.json")
    Observable<Message> deleteAllUserMessages(@Path("userId") String userId);
}

File: Application/src/main/kotlin/com/shareyourproxy/api/service/MessageService.kt
package com.shareyourproxy.api.service


import com.shareyourproxy.api.domain.model.Message
import retrofit.http.*
import rx.Observable
import java.util.*

/**
 * Service to get and update messages.
 */
interface MessageService {
    @GET("/messages/{userId}.json")
    fun getUserMessages(@Path("userId") userId: String): Observable<HashMap<String, Message>>

    @PUT("/messages/{userId}.json")
    fun addUserMessage(
            @Path("userId") userId: String, @Body message: HashMap<String, Message>): Observable<HashMap<String, Message>>

    @DELETE("/messages/{userId}.json")
    fun deleteAllUserMessages(@Path("userId") userId: String): Observable<Message>
}
